### ISSUE TITLE
New RDS DBCluster rules

### ISSUE THREAD

**Issue by kddejong** [aws-cloudformation/cfn-lint#4070]
### Describe the feature you'd like to request

When creating a DBCluster here are a set of issues I would like rules for:

  # The parameter MasterUsername must be provided and must not be blank
  # MasterUsername admin cannot be used as it is a reserved word used by the engine
  # If you specify storageEncrypted for instance that will be part of cluster it should match cluster setting; requested: true, cluster: false

### Describe the solution you'd like

I would like to detect these issues as best as I can.

### Additional context

```yaml

Resources:  
   # The parameter MasterUsername must be provided and must not be blank
  AuroraPgsqlDBCluster:
    Type: AWS::RDS::DBCluster
    Properties:
      Port: 5432
      Engine: aurora-postgresql
      MasterUserPassword: Password1!
      StorageEncrypted: false

  # If you specify storageEncrypted for instance that will be part of cluster it should match cluster setting; requested: true, cluster: false
  AuroraPgsqlDBMaster:
    Type: AWS::RDS::DBInstance
    Properties:
      DBInstanceClass: db.r5.xlarge
      DBClusterIdentifier: !Ref AuroraPgsqlDBCluster
      Engine: aurora-postgresql
      AutoMinorVersionUpgrade: true
      PubliclyAccessible: true
      StorageEncrypted: true

  # MasterUsername admin cannot be used as it is a reserved word used by the engine
  AuroraPgsqlDBClusterAdmin:
    Type: AWS::RDS::DBCluster
    Properties:
      Port: 5432
      Engine: aurora-postgresql
      MasterUsername: admin
      MasterUserPassword: Password1!
      StorageEncrypted: false

```

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request

### Would this feature include a breaking change?

- [ ] ⚠️ This feature might incur a breaking change
